src/webapp/components/current-data-submission/Validations.tsx

```diff
@@ -1,24 +1,104 @@
-import React from "react";
+import React, { useCallback, useState } from "react";
 import { EmbeddedReport } from "../reports/EmbeddedReport";
 import { CircularProgress } from "material-ui";
 import { useGlassDashboard } from "../../hooks/useGlassDashboard";
 import { Typography } from "@material-ui/core";
 import { useGetLastSuccessfulAnalyticsRunTime } from "../../hooks/useGetLastSuccessfulAnalyticsRunTime";
+import { useAppContext } from "../../contexts/app-context";
+import { useFileTypeByDataSubmission } from "../../hooks/useFileTypeByDataSubmission";
+import { useCurrentOrgUnitContext } from "../../contexts/current-orgUnit-context";
+import { useCurrentModuleContext } from "../../contexts/current-module-context";
+import { useCurrentPeriodContext } from "../../contexts/current-period-context";
+import { moduleProperties } from "../../../domain/utils/ModuleProperties";
+import { FileType } from "../../../domain/usecases/data-entry/DownloadTemplateUseCase";
+import moment from "moment";
+import styled from "styled-components";
+import { ContentLoader } from "../content-loader/ContentLoader";
+import { DownloadButton } from "./DownloadButton";
 
 export const Validations: React.FC = () => {
+    const [isLoading, setIsLoading] = useState(false);
+
     const { validationDashboardId } = useGlassDashboard();
     const { lastSuccessfulAnalyticsRunTime } = useGetLastSuccessfulAnalyticsRunTime();
+    const { api, compositionRoot } = useAppContext();
+    const fileType = useFileTypeByDataSubmission();
+
+    const { currentOrgUnitAccess } = useCurrentOrgUnitContext();
+    const { currentModuleAccess } = useCurrentModuleContext();
+    const { currentPeriod } = useCurrentPeriodContext();
+
+    const downloadTemplate = useCallback(
+        async (downloadType: "SUBMITTED" | "CALCULATED") => {
```

Comment:
This code is applicable only in AMC module, currently validation reports in other modules are not working. It continuously shows a content loader.

Reply:
Should be fixed now